src/registrar/management/commands/transfer_transition_domains_to_domains.py

```diff
@@ -0,0 +1,323 @@
+            {termColors.ENDC}
+            """
+        )
+        if len(skipped_domain_entries) > 0:
+            logger.info(
+                f"""{termColors.FAIL}
+                ============= SKIPPED DOMAINS (ERRORS) ===============
+                {skipped_domain_entries}
+                {termColors.ENDC}
+                """
+            )
+        if len(skipped_domain_invitations) > 0:
+            logger.info(
+                f"""{termColors.FAIL}
+                ============= SKIPPED DOMAIN INVITATIONS (ERRORS) ===============
+                {skipped_domain_invitations}
+                {termColors.ENDC}
+                """
+            )
+
+        # DEBUG:
+        self.print_debug(
+            debug_on,
+            f"""{termColors.YELLOW}
+
+            Created Domains:
+            {domains_to_create}
+
+            Updated Domains:
+            {updated_domain_entries}
+
+            {termColors.ENDC}
+            """,
+        )
+
+    def handle(
+        self,
+        **options,
+    ):
+        """Parse entries in TransitionDomain table
+        and create (or update) corresponding entries in the
+        Domain and DomainInvitation tables."""
+
+        # grab command line arguments and store locally...
+        debug_on = options.get("debug")
+        debug_max_entries_to_parse = int(
+            options.get("limitParse")
+        )  # set to 0 to parse all entries
+
+        self.print_debug_mode_statements(debug_on, debug_max_entries_to_parse)
+
+        # domains to ADD
+        domains_to_create = []
+        domain_invitations_to_create = []
+        # domains we UPDATED
+        updated_domain_entries = []
+        # domains we SKIPPED
+        skipped_domain_entries = []
+        # domainInvitations we SKIPPED
+        skipped_domain_invitations = []
+        # if we are limiting our parse (for testing purposes, keep
+        # track of total rows parsed)
+        total_rows_parsed = 0
+
+        logger.info(
+            f"""{termColors.OKGREEN}
+            ==========================
+            Beginning Data Transfer
+            ==========================
+            {termColors.ENDC}"""
+        )
+
+        for transition_domain in TransitionDomain.objects.all():
+            transition_domain_name = transition_domain.domain_name
+            transition_domain_status = transition_domain.status
+            transition_domain_email = transition_domain.username
+
+            # Check for existing domain entry
+            try:
+                # DEBUG:
+                self.print_debug(
+                    debug_on,
+                    f"""{termColors.OKCYAN}
+                    Processing Transition Domain: {transition_domain_name}, {transition_domain_status}, {transition_domain_email}
+                    {termColors.ENDC}""",  # noqa
+                )
+
+                # get the existing domain
+                target_domain = Domain.objects.get(name=transition_domain_name)
```

Comment:
Why are we not using .exists() here?

Reply:
That would be the proper way to do this.  I think part of me thought it was less efficient, checking if an object exists and then grabbing it is traversing the table twice...and makes this function more complex...plus we check for other exceptions anyways.  But I agree this isn't proper form so I will change it.